src/goto-symex/symex_goto.cpp

```diff
@@ -221,7 +221,16 @@ void goto_symext::symex_goto(statet &state)
   framet::goto_state_listt &goto_state_list =
     state.top().goto_state_map[new_state_pc];
 
-  goto_state_list.emplace_back(state);
+  // If the guard is true (and the alternative branch unreachable) we can move
```

Comment:
But ... we pass `state` to `symex_transition` just below?!

Reply:
This suggests a constructor taking an ordinary ref and a boolean `can_move_state` or something, which partially destroys `goto_symex_statet` (but critically leaves its source and stats unchanged / suitable for continued usage)?

Reply:
I've added this change in as an additional commit, please check it when able as I wasn't sure what fields were safe to move or not.